src/app/style_helpers.ts

```diff
@@ -1,87 +1,119 @@
-type Filter = {filter: FilterSpecification};
+type Layer = LineLayerSpecification & { source: string };
+type Filter = { filter: FilterSpecification };
 type Cap = PropertyValueSpecification<"round">;
-type ColorMaps = {[Status in TronçonStatus]: string};
-type MaplibreColor = {"line-color": DataDrivenPropertyValueSpecification<ColorSpecification>};
-type Visbility = {"line-opacity": DataDrivenPropertyValueSpecification<number>};
-type LineWidth = {"line-width"?: DataDrivenPropertyValueSpecification<number>};
+type ColorMaps = { [Status in TronçonStatus]: string };
+type MaplibreColor = {
+  "line-color": DataDrivenPropertyValueSpecification<ColorSpecification>;
+};
+type Visbility = {
+  "line-opacity": DataDrivenPropertyValueSpecification<number>;
+};
+type LineWidth = {
+  "line-width"?: DataDrivenPropertyValueSpecification<number>;
+};
 
 // Very basic boilerplate to create a line layer
 // Use it as addLayer({...baselayer('id'), paint: {...}, filter: {....}})
 export function baseLayer(id: string, setCap: boolean = true): Layer {
-  const caps = setCap ? {
-    'line-cap': 'round' as Cap,
-    'line-join': 'round' as Cap,
-  } : {}
+  const caps = setCap
+    ? {
+        "line-cap": "round" as Cap,
+        "line-join": "round" as Cap,
+      }
+    : {};
 
   return {
     id,
-    source: 'vif',
-    type: 'line',
+    source: "vif",
+    type: "line",
     layout: {
-        ...caps,
-      'line-sort-key': ['get', ['get', 'status'], ['literal', statusIndex]],
+      ...caps,
+      "line-sort-key": ["get", ["get", "status"], ["literal", statusIndex]],
     },
-  }
+  };
 }
 
 export const onlyVariants: Filter = { filter: ["get", "variant"] };
-export const exceptedVariants: Filter = { filter: ["!", ["get", "variant"]]};
+export const exceptedVariants: Filter = { filter: ["!", ["get", "variant"]] };
 
 export function colorFromStatus(colors: ColorMaps): MaplibreColor {
-    return { "line-color": [ "get", ["get", "status"], ["literal", colors] ]
-}}
+  return { "line-color": ["get", ["get", "status"], ["literal", colors]] };
+}
 
 export function hideWhen(state: string): Visbility {
   return {
     "line-opacity": [
-        "case",
-        ["boolean", ["feature-state", state], false],
-        0.0,
-        1,
+      "case",
+      ["boolean", ["feature-state", state], false],
+      0.0,
+      1,
     ],
-  }
+  };
 }
 
 export function showWhen(state: string): Visbility {
- return {
+  return {
     "line-opacity": [
-        "case",
-        ["boolean", ["feature-state", state], false],
-        1,
-        0.0,
+      "case",
+      ["boolean", ["feature-state", state], false],
+      1,
+      0.0,
     ],
-  }
+  };
 }
 
 export function width(widthAtZoom10: number, widthAtZoom15: number): LineWidth {
   return {
     "line-width": [
-      "interpolate", ["linear"], ["zoom"],
-      10, widthAtZoom10,
-      15, widthAtZoom15,
+      "interpolate",
+      ["linear"],
+      ["zoom"],
+      10,
```

Comment:
ici on aurait pu desactiver le linter, mais flemme. Ship it

Reply:
Ah, zut, j’ai tiré trop vite. Il faudrait sans doute des règles pour prettify spécifiques à maplibre-style-spec.